frontend/javascripts/oxalis/model/volumetracing/volume_annotation_sampling.js

```diff
@@ -333,24 +345,269 @@ export function applyVoxelMap(
         continue;
       }
       const { data } = bucket.getOrCreateData();
+      // TODO: In case a segment part that already exists in the backend and the bucket is overdrawn,
+      // but the bucket data isn't fetched until this step, we do not know that this segment part is overdrawn.
+      // This breaks the housekeeping of the covered buckets by a segment.
+      //
+      // A possible solution would be that when the incoming backend data is merged with the frontend data,
+      // the merging automatically updates the covered buckets. The merge operation could use the
+      // manageRemovingBucketAddressesOfOverdrawnSegments-method to do that. Then it only needs to track which segments were overwritten.
+      //
+      // ^ This is already done in bucket.js - merge. Look there for more details
       for (let firstDim = 0; firstDim < constants.BUCKET_WIDTH; firstDim++) {
         for (let secondDim = 0; secondDim < constants.BUCKET_WIDTH; secondDim++) {
           if (voxelMap[firstDim * constants.BUCKET_WIDTH + secondDim] === 1) {
+            annotatedBucket = isBucketInLowestResolution;
             get3DAddress(firstDim, secondDim, out);
             const voxelToLabel = out;
             voxelToLabel[thirdDimensionIndex] =
               (voxelToLabel[thirdDimensionIndex] + sliceCount) % constants.BUCKET_WIDTH;
             // The voxelToLabel is already within the bucket and in the correct resolution.
             const voxelAddress = dataCube.getVoxelIndexByVoxelOffset(voxelToLabel);
-            if (shouldOverwrite || (!shouldOverwrite && data[voxelAddress] === overwritableValue)) {
+            const currentSegmentId = data[voxelAddress];
+            if (shouldOverwrite || currentSegmentId === overwritableValue) {
+              if (
+                isBucketInLowestResolution &&
+                currentSegmentId > 0 &&
```

Comment:
This line should be removed, as erasing (overwriting with id 0) also overwrites the segment id at the current position (`voxelAddress`).